Assets/Plugins/Source/Core/PlatformManager.cs

```diff
@@ -0,0 +1,294 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using Settings;
+    using System.Collections.Generic;
+    using System;
+    using System.IO;
+
+#if UNITY_EDITOR
+    using UnityEditor;
+
+#endif
+
+    using UnityEngine;
+
+    public static class PlatformManager
+    {
+        /// <summary>
+        /// Enum that stores the possible platforms
+        /// </summary>
+        public enum Platform
+        {
+            Unknown,
+            Windows,
+            Android,
+            XboxOne,
+            XboxSeriesX,
+            iOS,
+            Linux,
+            macOS,
+            PS4,
+            PS5,
+            Switch,
+            Steam
+        }
+
+        private struct PlatformInfo
+        {
+            public string FullName;
+            public string ConfigFileName;
+            public Type ConfigType;
+            public string DynamicLibraryExtension;
+        }
+
+        /// <summary>
+        /// Private collection to store associations between a platform type, it's human readable name, the file in which the configuration is stored, and the type of the config object
+        /// </summary>
+        private static IDictionary<Platform, PlatformInfo> PlatformInformation =
+            new Dictionary<Platform, PlatformInfo>();
+
+        /// <summary>
+        /// Backing value for the CurrentPlatform property.
+        /// </summary>
+        private static Platform s_CurrentPlatform;
+
+        /// <summary>
+        /// Returns the current platform. In-order to reduce the number of places in the build pipeline
+        /// where build defines are if-branched, this value is designed to be set from outside of the
+        /// static class "PlatformManager."
+        /// </summary>
+        public static Platform CurrentPlatform
+        {
+            get { return s_CurrentPlatform; }
+            set
+            {
+                // This is to ensure that the platform is only ever determined once
+                if (CurrentPlatform == Platform.Unknown)
+                    s_CurrentPlatform = value;
+            }
+        }
+
+        static PlatformManager()
+        {
+            AddPlatformInfo(Platform.Android, "Android", "eos_android_config.json", typeof(AndroidConfig), ".so");
+            AddPlatformInfo(Platform.iOS, "iOS", "eos_ios_config.json", typeof(IOSConfig), ".dylib");
+            AddPlatformInfo(Platform.Linux, "Linux", "eos_linux_config.json", typeof(LinuxConfig), ".so");
+            AddPlatformInfo(Platform.macOS, "macOS", "eos_macos_config.json", typeof(MacOSConfig), ".dylib");
+            AddPlatformInfo(Platform.Steam, "Steam", "eos_steam_config.json", typeof(EOSSteamConfig), "");
```

Comment:
Confirm that `.so` is correct for Android.

Reply:
Yes - it's correct - but irrelevant, because Android doesn't have native code that needs to be compiled.